common/logic.proto

```diff
@@ -106,3 +109,14 @@ message Rule {
         message Res {}
     }
 }
+
+message Explanation {
+    Rule rule = 1;
+    map<string, VarsList> varMapping = 2; // TODO perhaps this should be a "context" and contain more information about source
```

Comment:
We need to use `snake_case` consistently everywhere for protocol field names - Java handles mixed casing relatively gracefully, but Python and JS don't.
```suggestion
    map<string, VarsList> var_mapping = 2; // TODO perhaps this should be a "context" and contain more information about source
```
(Other examples elsewhere in the PR)